### PR TITLE
nrf: General cleanup

### PR DESCRIPTION
**Pull request by arturo182**
A bit of cleanup before updating the nRF HAL to a newer version.

### CODE REVIEW

File: ports/nrf/Makefile [hunk 1]
```diff
@@ -12,24 +12,17 @@ SD_LOWER = $(shell echo $(SD) | tr '[:upper:]' '[:lower:]')
...
 # TODO: Verify that it is a valid target.
 
-ifeq ($(SD), )
-	# If the build directory is not given, make it reflect the board name.
-	BUILD ?= build-$(BOARD)
-	include ../../py/mkenv.mk
-	include boards/$(BOARD)/mpconfigboard.mk
-	-include mpconfigport.mk
+# If the build directory is not given, make it reflect the board name
+BUILD ?= $(if $(SD),build-$(BOARD)-$(SD_LOWER),build-$(BOARD))
```
Comment (tannewt) [inline ports/nrf/Makefile]: Do we really need the softdevice here? my understanding is that its fixed per board.

Comment (arturo182) [reply ports/nrf/Makefile]: I think it's useful if you want to build with and without SD without having to remove the build directory, I was actually tempted to also add the SD version in there so we could build 2.0.1 and 5.0.0 without cleaning but I think there are some .mk file dependencies that make it hard.

**Comment by arturo182** [comment]
@tannewt Oh, I was not aware there was any MP PR for nRF, I'll look into it.

@jerryneedell I missed that, I will like that and add to this PR, thanks!

**Comment by arturo182** [comment]
@tannewt Just added one small change, you approved the previous commits so this shouldn't take long to approve and merge :D

File: ports/nrf/Makefile (hunk 1 above)
Comment (hathach) [reply ports/nrf/Makefile]: SD is upgradable, it is useful, we will role out the bootloader with S132 v6 for feather52832, to be consistent with nrf52840

**Pull request merged**